test/v1/controllers/requestParsers/validators/validations/GivenNameValidationSpec.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers.validators.validations
+
+import org.scalatest.Inspectors
+import support.UnitSpec
+
+// Note that these tests are sanity checks of the implementation of a specified regex rather than being definitive.
+class GivenNameValidationSpec extends UnitSpec with Inspectors {
+  val validator: GivenNameValidation.type = GivenNameValidation
+
+  "GivenNameValidation" must {
+    "allow alphanumeric values with arbitrary capitalization" in {
+      forAll(Seq("alice", "Bob")) { name =>
+        validator.isValid(name) shouldBe true
+      }
+    }
+
+    "allow hyphenated and spaces and dots" in {
+      forAll(Seq("Tommy-Lee", "Mary Jane", "will.i")) { name =>
+        validator.isValid(name) shouldBe true
+      }
+    }
+
+    // Re-instate if reqd once spec finalized
+    //    "allow diacritics" in {
+    //      forAll(Seq("Arsène", "José", "Jokūbas", "Jørgen")) { name =>
+    //        validator.isValid(name) shouldBe true
+    //      }
+    //    }
+
+    "allow ordinals" in {
+      forAll(Seq("Henry 8th")) { name =>
+        validator.isValid(name) shouldBe true
+      }
+    }
+
+    "disallow special characters and emojis" in {
+      forAll(Seq("{}", "_", "\uD83D\uDC4D", "; drop table students; --")) { name =>
+        validator.isValid(name) shouldBe false
+      }
+    }
+
+    "allow values at both ends of length limit" in {
+      forAll(Seq("A", "a".repeat(35))) { name =>
```

Comment:
my inteliji really doesn't seem to like ".repeat" and just refuses to compile it